src/leer.java

```diff
@@ -8,264 +8,210 @@
+		    catch (Exception e) {
+                  e.printStackTrace();
+			}
+        }
+	}
+public  double calculate_speed (String lat,
+						  String lng,
+						  String lat_line2,
+						  String lng_line2,
+						  String id_ride,
+						  String timestamp,
+						  String timestamp_line2,
+						  int i)
+{
+	double a,c,distance,U,delta_latitude,delta_longitude,delta_time;
 	int R =6371;
+
 	leer calculate = new leer();
 	
-    float id_ridea = Float.parseFloat(id_ride);
-    float lata = Float.parseFloat(lat);
-    float latb = Float.parseFloat(lat1);
-    float lnga = Float.parseFloat(lng);
-    float lngb = Float.parseFloat(lng1);
-    Integer timestampa = Integer.parseInt(timestamp);
-    Integer timestampb = Integer.parseInt(timestamp1);
-	
-//calculating distance
-    deltlat=Math.toRadians(latb - lata) ;
-    deltlng= Math.toRadians(lngb - lnga);
-    delttime= (timestampb - timestampa);
-    a = Math.pow(Math.sin(deltlat/2),2)+ Math.cos(lata)*Math.cos(latb)* Math.pow(Math.sin(deltlng/2),2);
-    c = 2* Math.atan2(((Math.sqrt(a))),(Math.sqrt(1-a)));
+    float new_id_ride = Float.parseFloat(id_ride);
+    float new_lat = Float.parseFloat(lat);
+    float new_lat_line2 = Float.parseFloat(lat_line2);
+    float new_lng = Float.parseFloat(lng);
+    float new_lng_line2 = Float.parseFloat(lng_line2);
+    int new_timestamp = Integer.parseInt(timestamp);
+    int new_timestamp_line2 = Integer.parseInt(timestamp_line2);
+
+    delta_latitude = Math.toRadians(new_lat_line2 - new_lat) ;
+    delta_longitude = Math.toRadians(new_lng_line2 - new_lng);
+    delta_time = (new_timestamp_line2 - new_timestamp);
+    a = Math.pow(Math.sin(delta_latitude/2),2) +
+			Math.cos(new_lat) * Math.cos(new_lat_line2) *
+					Math.pow(Math.sin(delta_longitude/2),2);
+	c = 2 * Math.atan2(((Math.sqrt(a))),(Math.sqrt(1-a)));
     distance = (R*c);
-    U = (distance/delttime)*3600;
-    
-	calculate.fare_rules(distance, U, deltlat, deltlng, delttime, id_ridea, timestampa, i);
+	U = (distance/delta_time)*3600;
+	calculate.fare_rules(distance, U, delta_time, new_id_ride, new_timestamp, i);
 	
     return U;
 }
+public static String[] filter (String[] filtered_fields, double U) {
 
-public static String[] filter (String[] camposa, double U) {
-	
-	//filter
-	
 	if (U > 100) {
-		  
-	      camposa[0]= "0" ;
-	      camposa[1]= "0";
-	      camposa[2]= "0";
-	      camposa[3]= "0";
-       //System.out.println("AAAAA");      
-     }
-	
-	return camposa; 
+	      filtered_fields[0]= "0" ;
+	      filtered_fields[1]= "0";
+	      filtered_fields[2]= "0";
+	      filtered_fields[3]= "0";
+	}
 	
+	return filtered_fields;
 }
+public  void fare_rules( double distance,double U,double delta_time,float new_id_ride, int new_timestamp, int i) {
 
-public  void fare_rules( double distance,double U,double deltlat,double deltlng,double delttime,float id_ridea, int timestampa, int i) {
-	// Fare rules
 	double  fare_amount;
-    
+	// when the speed is less than 10KM
```

Comment:
I like this comment, because explain why  <10

Reply:
great! i will add more!!